numpy/core/src/multiarray/lowlevel_strided_loops.c.src

```diff
@@ -1666,10 +1667,12 @@ mapiter_@name@(PyArrayMapIterObject *mit)
         }
 
         counter = NpyIter_GetInnerLoopSizePtr(mit->subspace_iter);
-        if (*counter == PyArray_SIZE(mit->subspace)) {
-            skip = 1;
+        if (*counter != PyArray_SIZE(mit->subspace)) {
+           /*
+            * if the subspace iterator skips, we cannot avoid resetting
+            */
```

Comment:
Remembered the word, its "trivially iterable" (at least within much of our code). Just trying to make it a bit clearer (only some thougths): If the subspace iterator is not trivially iterable (so not in a single stride), it has to be reset to the correct start point in every outer iteration. If it is trivially iterable we can avoid using it alltogether (the actual loop does nothing).